Project/Controllers/UserProfileController.cs

```diff
@@ -188,35 +213,41 @@ public class UserProfileController : Controller {
         [HttpPost]
         [Authorize, ValidateAntiForgeryToken]
         [Route("AddStory")]
-        public async Task<ActionResult> AddStory(StoryVM story) {
-
-            if (!ModelState.IsValid) {
+        public async Task<ActionResult> AddStory(StoryVM story)
+        {
+            if (!ModelState.IsValid)
+            {
                 return PartialView("_AjaxValidation", "Required story fields were not filled in.");
             }
 
             var storyModel = Mapper.Map<Story>(story);
-            storyModel.Id = 0;
+
             storyModel.State = Core.Models.ModelState.Added;
+            storyModel.Date = DateTime.Now;
+            UserInfo user = await userService.FindUserByIdAsync(storyModel.UserId);
```

Comment:
This is not used. Also, the userId should be retrieved with the userService.GetUserId() because the story owner should be the currently logged in user. Setting the NewStory.UserId in the Index() method is also redundant.

Reply:
I put that before deleting the User from Story model.